website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,83 @@
 # Widget Errors
 
-This section will help you troubleshoot common widget errors on the Appsmith platform.
+This section would help you troubleshoot common widget errors on the Appsmith platform.
 
-## JSON Form Errors
+## Execution errors
+You may see below errors when executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/).
+
+### Sync fields errors
+
+You may see this error when executing an API/Query/JS Object from a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+You would see one of the below error messages based on API/Query or a JS Object function execution.
+
+> Found a reference to {{action}} during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to {{action}} and try again.
+
+ or 
+
+> Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.
+
+
+**To-Do to add screenshot**
+
+#### Cause
+Action refers to the execution of an API/Query/JS object. 
```

Comment:
nitpick . For some reason, the / is not looking good. 

Wherever this is being done, can we write it with a comma as
API, Query, or JS Object.